### PR TITLE
Added .travis.yml for Travis CI (PS-5.7)

### PR DESCRIPTION
**Pull request by inikep**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  - CC=$CC_COMPILER CXX=$CXX_COMPILER cmake ..
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    -DMYSQL_MAINTAINER_MODE=ON
+    -DBUILD_CONFIG=mysql_release
+    -DFEATURE_SET=community
+    -DENABLE_DTRACE=OFF
+    -DWITH_SSL=$SSL
+    -DWITH_ZLIB=system
+    -DENABLE_DOWNLOADS=1
+    -DWITHOUT_TOKUDB=ON
```
Comment (percona-ysorokin) [inline .travis.yml]: The same as in 5.6 (both TokuDB and RocksDB must be enabled).

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  - CC=$CC_COMPILER CXX=$CXX_COMPILER cmake ..
+    -DCMAKE_BUILD_TYPE=Debug
+    -DMYSQL_MAINTAINER_MODE=ON
+    -DBUILD_CONFIG=mysql_release
+    -DFEATURE_SET=community
+    -DENABLE_DTRACE=OFF
+    -DWITH_SSL=$SSL
+    -DWITH_ZLIB=system
+    -DENABLE_DOWNLOADS=1
+    -DWITHOUT_TOKUDB=ON
```
Comment (percona-ysorokin) [inline .travis.yml]: The same as in 5.6 (both TokuDB and RocksDB must be enabled).

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

**Comment by RoelVdP** [comment]
@percona-ysorokin maybe I am misreading something here, but the comment says "both TokuDB and RocksDB must be enabled" whereas the RocksDB build option is -DWITHOUT_ROCKSDB=ON instead of -DWITH_ROCKSDB and TokuDB has -DWITHOUT_TOKUDB=ON as well?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+  # Test "RelWithDebInfo" compilation
+  - CC=$CC_COMPILER CXX=$CXX_COMPILER cmake ..
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    -DMYSQL_MAINTAINER_MODE=ON
+    -DBUILD_CONFIG=mysql_release
+    -DFEATURE_SET=community
+    -DENABLE_DTRACE=OFF
+    -DWITH_SSL=$SSL
+    -DWITH_ZLIB=system
```
Comment (laurynas-biveinis) [inline .travis.yml]: -DWITH_CURL=system -DWITH_LZ4=system -DWITH_MECAB=system -DWITH_PROTOBUF=system ?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  # Test "Debug" compilation
+  - rm -rf *
+  - CC=$CC_COMPILER CXX=$CXX_COMPILER cmake ..
+    -DCMAKE_BUILD_TYPE=Debug
+    -DMYSQL_MAINTAINER_MODE=ON
+    -DBUILD_CONFIG=mysql_release
+    -DFEATURE_SET=community
+    -DENABLE_DTRACE=OFF
+    -DWITH_SSL=$SSL
+    -DWITH_ZLIB=system
```
Comment (laurynas-biveinis) [inline .travis.yml]: Likewise

**Comment by percona-ysorokin** [comment]
@RoelVdP You are absolutely right. The fix is on its way.

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Comment by RoelVdP** [comment]
@percona-ysorokin hmm so maybe I am not following something here, but the merge includes;

```
+    -DWITHOUT_TOKUDB=ON
+    -DWITHOUT_ROCKSDB=ON
```

i.e. both are disabled?

**Comment by percona-ysorokin** [comment]
@RoelVdP Unfortunately, we encountered a limitation with free Travis-CI accounts. Any job cannot last longer than 50 min. And we were far beyond that limit when compiling full source tree. So, we had to TEMPORARILY disable both toku and rocks engines for 5.7.
We are not giving up and still investigating how we can improve compilation speed. So, just be patient. this is gonna be an iterative process.

**Comment by RoelVdP** [comment]
Got it, thanks

**Pull request merged**